core/src/debug_ui/movie.rs

```diff
@@ -0,0 +1,320 @@
+
+                ui.label("Size");
+                ui.label(format!("{} x {}", movie.width(), movie.height()));
+                ui.end_row();
+
+                ui.label("Frame Rate");
+                ui.label(movie.frame_rate().to_string());
+                ui.end_row();
+
+                ui.label("Compression");
+                ui.label(format!("{:?}", movie.header().compression()));
+                ui.end_row();
+
+                ui.label("Length");
+                ui.label(format!(
+                    "{} bytes ({} compressed)",
+                    movie.uncompressed_len(),
+                    movie.compressed_len()
+                ));
+                ui.end_row();
+
+                ui.label("Num Frames");
+                ui.label(movie.num_frames().to_string());
+                ui.end_row();
+
+                ui.label("Flags");
+                ui.add_enabled_ui(false, |ui| {
+                    ui.vertical(|ui| {
+                        if movie.header().is_action_script_3() {
+                            ui.label("Uses Actionscript 3");
+                        }
+                        if movie.header().has_metdata() {
+                            ui.label("Has XMP Metadata");
+                        }
+                        if movie.header().use_direct_blit() {
+                            ui.label("Use Direct Blit");
+                        }
+                        if movie.header().use_gpu() {
+                            ui.label("Use GPU");
+                        }
+                        if movie.header().use_network_sandbox() {
+                            ui.label("Use Network Sandbox");
+                        }
+                    })
+                });
+                ui.end_row();
+            });
+
+        if !movie.parameters().is_empty() {
+            CollapsingHeader::new("Parameters")
+                .id_source(ui.id().with("parameters"))
+                .default_open(false)
+                .show(ui, |ui| {
+                    Grid::new(ui.id().with("parameters"))
+                        .num_columns(2)
+                        .show(ui, |ui| {
+                            for (key, value) in movie.parameters() {
+                                ui.text_edit_singleline(&mut key.as_str());
+                                ui.text_edit_singleline(&mut value.as_str());
+                                ui.end_row();
+                            }
+                        });
+                });
+        }
+    }
+}
+
+pub fn movie_name(movie: &Arc<SwfMovie>) -> String {
+    format!("SWF {:p}", Arc::as_ptr(movie))
+}
+
+pub fn open_movie_button(ui: &mut Ui, movie: &Arc<SwfMovie>, messages: &mut Vec<Message>) {
+    if ui.button(movie_name(movie)).clicked() {
+        messages.push(Message::TrackMovie(movie.clone()));
+    }
+}
+
+pub fn open_character_button(ui: &mut Ui, character: &Character) {
+    match character {
+        Character::EditText(_) => {
```

Comment:
do you expect some more complex per-type logic here? Otherwise, how about a much simpler enum->string literal match, and a single `label()` call at the end?